features/guidance/notification.feature

```diff
@@ -0,0 +1,65 @@
+            | b | c |   |   | d | e |
+            |   |   |   |   |   |   |
+            | a |   |   |   |   | g |
+
+        And the ways
+            | nodes | highway | route | name  |
+            | ab    | primary |       | bingo |
+            | bc    | primary |       | dingo |
+            | cde   |         | ferry | ferry |
+            | eg    | primary |       | wingo |
+
+        When I route I should get
+            | waypoints | route                   | turns                                                   | modes                               |
+            | a,g       | bingo,ferry,wingo,wingo | depart,notification straight,notification right,arrive  | driving,ferry,driving,driving       |
+
+    Scenario: Turn onto a Ferry
+        Given the node map
+            |   |   |   |   |   |   | g |
+            |   | a |   |   |   |   | e |
+            |   |   | c |   |   | d |   |
+            |   |   |   |   |   |   | f |
+            |   | b |   |   |   |   |   |
+
+        And the ways
+            | nodes | highway | route | name  | oneway |
+            | ca    | primary |       | bingo | yes    |
+            | bc    | primary |       | dingo | yes    |
+            | cde   |         | ferry | ferry | no     |
+            | cdf   |         | ferry | ferry | no     |
+            | ge    | primary |       | wingo | no     |
+
+        When I route I should get
+            | waypoints | route                         | turns                                                               | modes                                   |
+            | g,a       | wingo,ferry,ferry,bingo,bingo | depart,notification right,continue right,notification right,arrive  | driving,ferry,ferry,driving,driving     |
+            | b,g       | dingo,ferry,ferry,wingo,wingo | depart,turn right,continue left,notification left,arrive            | driving,ferry,ferry,driving,driving     |
+
+    Scenario: Straight onto a Ferry
+        Given the node map
+            |   |   |   |   |   |   |   |   |
+            | a | b | c |   |   | d |   | i |
+            |   |   |   |   |   |   |   |   |
+            |   |   |   |   |   |   | f |   |
+
+        And the ways
+            | nodes | highway | route | name    | oneway |
+            | ab    | primary |       | yonkers | yes    |
+            | bc    | primary |       | dingo   | yes    |
+            | ca    | primary |       | bingo   | yes    |
+            | dc    |         | ferry | ferry   | no     |
+            | df    | primary |       | maple   | yes    |
+            | fi    | primary |       | dolly   | yes    |
+
+        When I route I should get
+            | waypoints | route                            | turns                                                              | modes                                 |
+            | a,i       | yonkers,ferry,maple,dolly,dolly  | depart,turn straight,notification right,new name sharp left,arrive | driving,ferry,driving,driving,driving |
```

Comment:
so you are getting a turn-straight here. I don't know if this is being solved already by my changes in the turn-angles branch, but going straight here might be considered obvious, which should be a notification in this case.